### PR TITLE
New recipe sparc-x-api

### PR DESCRIPTION
**Pull request by alchem0x2A**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by alchem0x2A** [comment]
@conda-forge/staged-recipes @conda-forge/help-python

**Comment by alchem0x2A** [comment]
@conda-forge/staged-recipes @conda-forge/help-python please review & merge the PR once you have time, thx!

**Comment by alchem0x2A** [comment]
@conda-forge/staged-recipes @conda-forge/help-python pinging again for review request. Thank you!

File: recipes/sparc-x-api/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{% set name = "SPARC-X-API" %}
+{% set version = "1.0.0" %}
+{% set file_ext = ".tar.gz" %}
```
Comment (ocefpaf) [inline recipes/sparc-x-api/meta.yaml:3]:
```suggestion
```

File: recipes/sparc-x-api/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
```
Comment (ocefpaf) [inline recipes/sparc-x-api/meta.yaml:8]:
```suggestion
  name: {{ name|lower }}
  version: {{ version }}
```

File: recipes/sparc-x-api/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+source:
+  url: https://github.com/SPARC-X/{{ name }}/archive/refs/tags/v{{ version }}{{ file_ext }}
```
Comment (ocefpaf) [inline recipes/sparc-x-api/meta.yaml:11]:
```suggestion
  url: https://github.com/SPARC-X/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
```

File: recipes/sparc-x-api/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  sha256: 34aa3a8d0801a91736b0d8322d78d1e432e92585e61af54db3aba17340ff7617
+
+build:
+  noarch: python
+  number: 0
+  # Install and include the psp files
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv && cd .. && {{ PYTHON }} -m sparc.download_data"
+  entry_points:
+    - "sparc-ase = sparc.cli:main"
+
```
Comment (ocefpaf) [inline recipes/sparc-x-api/meta.yaml:21]:
```suggestion
```

File: recipes/sparc-x-api/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  entry_points:
+    - "sparc-ase = sparc.cli:main"
+    
+
+requirements:
+  build:
+    - python >=3.8
+    - setuptools
+    - ase >=3.22
+    - pip
```
Comment (ocefpaf) [inline recipes/sparc-x-api/meta.yaml:28]: You don't need build and host for a noarch package.

```suggestion
```

File: recipes/sparc-x-api/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  host:
+    - python >=3.8
+    - ase >=3.22
```
Comment (ocefpaf) [inline recipes/sparc-x-api/meta.yaml:31]: You don't need runtime dependencies at build time. Unless the package imports itself at build time, then you should report this upstream.

```suggestion
```

File: recipes/sparc-x-api/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  license_family: GPL
+  license_file: LICENSE
+  summary: "SPARC-X-API: A Python API for the SPARC-X DFT Code"
+  description: |
+    `SPARC-X-API` is an [ASE](https://wiki.fysik.dtu.dk/ase/)-compatible Python API for the density functional theory (DFT) code [SPARC](https://github.com/SPARC-X/SPARC). It offers:
+    1. ASE-compatible I/O format for SPARC files
+    2. A JSON API interfacing with SPARC's C-code for parameter validation and conversion
+    3. A comprehensive calculator interface for SPARC.
+  doc_url: https://github.com/SPARC-X/{{ name }}/blob/master/README.md
+  dev_url: https://github.com/SPARC-X/{{ name }}
```
Comment (ocefpaf) [inline recipes/sparc-x-api/meta.yaml:60]: The jinja variable here obfuscates the URLs and makes them unclickable.

```suggestion
  home: https://github.com/SPARC-X/SPARC-X-API
  license: GPL-3.0-or-later
  license_family: GPL
  license_file: LICENSE
  summary: "SPARC-X-API: A Python API for the SPARC-X DFT Code"
  description: |
    `SPARC-X-API` is an [ASE](https://wiki.fysik.dtu.dk/ase/)-compatible Python API for the density functional theory (DFT) code [SPARC](https://github.com/SPARC-X/SPARC). It offers:
    1. ASE-compatible I/O format for SPARC files
    2. A JSON API interfacing with SPARC's C-code for parameter validation and conversion
    3. A comprehensive calculator interface for SPARC.
  doc_url: https://github.com/SPARC-X/SPARC-X-API/blob/master/README.md
  dev_url: https://github.com/SPARC-X/SPARC-X-API
```

File: recipes/sparc-x-api/meta.yaml (hunk 6 above)
Comment (alchem0x2A) [reply recipes/sparc-x-api/meta.yaml:24]: Indeed the line `script: "{{ PYTHON }} -m pip install . --no-deps -vv && cd .. && {{ PYTHON }} -m sparc.download_data"` uses a post-installation command to add additional binary data which requires imports. is this allowed in conda-forge or I should use alternative approaches?

Comment (ocefpaf) [reply recipes/sparc-x-api/meta.yaml:24]: Did you try removing it? If it is post installation you don't need it in build.
Note that conda also has [post link](https://docs.conda.io/projects/conda-build/en/stable/resources/link-scripts.html#adding-pre-link-post-link-and-pre-unlink-scripts) for this kind of thing if just removing the run-time dependency doesn't work.

Comment (alchem0x2A) [reply recipes/sparc-x-api/meta.yaml]: @ocefpaf post-link sounds a better option. I have updated the recipe with the postlink scripts and subsequently changed the tests. appreciate your feedback

**Comment by alchem0x2A** [comment]
@ocefpaf the updated recipe now uses post-link scripts. Appreciate your feedback

File: recipes/sparc-x-api/meta.yaml [hunk 8]
```diff
@@ -14,14 +14,13 @@ build:
...
   # Install and include the psp files
-  script: "{{ PYTHON }} -m pip install . --no-deps -vv && cd .. && {{ PYTHON }} -m sparc.download_data"
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
   entry_points:
     - "sparc-ase = sparc.cli:main"
 
 requirements:
   host:
     - python >=3.8
-    - ase >=3.22
```
Comment (ocefpaf) [inline recipes/sparc-x-api/meta.yaml]: Did you try just removing this first? B/c if that works you don't need the post link. Post links will require to remove noarch and that is not desirable.

File: recipes/sparc-x-api/meta.yaml [hunk 9]
```diff
@@ -8,20 +8,19 @@ package:
...
   # Install and include the psp files
-  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv && cd .. && {{ PYTHON }} -m sparc.download_data"
   entry_points:
     - "sparc-ase = sparc.cli:main"
 
 requirements:
   host:
     - python >=3.8
-    - pip
```
Comment (ocefpaf) [inline recipes/sparc-x-api/meta.yaml]: You do need `pip` here b/c we use it in the python pip install command above.

Comment (alchem0x2A) [reply recipes/sparc-x-api/meta.yaml]: @ocefpaf Sorry for that. I've updated the meta.yaml.

**Comment by alchem0x2A** [comment]
thx! @ocefpaf

**Pull request merged**